pkg/route/ingress/ingress.go

```diff
@@ -634,14 +647,22 @@ func routeMatchesIngress(
 	path *networkingv1.HTTPIngressPath,
 	secretLister corelisters.SecretLister,
 	serviceLister corelisters.ServiceLister,
+	host string,
```

Comment:
I'm wondering whether we need to carry this info from `sync()` to `newRouteForIngress()` while the later function has everything to decide about the wildcardPolicy on its own: `rule *networkingv1.IngressRule` parameter.

Something like `targetPortForService()` function used at the very beginning of `newRouteFromIngress` can be created for the host name and the wildcard policy:
```go
host, wildcardPolicy := hostAndWildcardPolicy(rule)
```
What do you think?